Userland/Applications/TextEditor/main.cpp

```diff
@@ -36,12 +36,16 @@ int main(int argc, char** argv)
     if (file_to_edit) {
         FileArgument parsed_argument(file_to_edit);
 
-        file_to_edit_full_path = Core::File::real_path_for(parsed_argument.filename());
-        VERIFY(!file_to_edit_full_path.is_empty());
-        dbgln("unveil for: {}", file_to_edit_full_path);
-        if (unveil(file_to_edit_full_path.characters(), "r") < 0) {
-            perror("unveil");
-            return 1;
+        if (Core::File::exists(parsed_argument.filename())) {
+            file_to_edit_full_path = Core::File::real_path_for(parsed_argument.filename());
+            VERIFY(!file_to_edit_full_path.is_empty());
+            dbgln("unveil for: {}", file_to_edit_full_path);
+            if (unveil(file_to_edit_full_path.characters(), "r") < 0) {
+                perror("unveil");
+                return 1;
+            }
+        } else {
+            file_to_edit_full_path = Core::File::absolute_path(parsed_argument.filename());
         }
```

Comment:
The `absolute_path()` method handles files that exist as well, how about something like
```suggestion
        file_to_edit_full_path = Core::File::absolute_path(parsed_argument.filename());
        VERIFY(!file_to_edit_full_path.is_empty());
        if (Core::File::exists(file_to_edit_full_path)) {
        	dbgln("unveil for: {}", file_to_edit_full_path);
            if (unveil(file_to_edit_full_path.characters(), "r") < 0) {
                perror("unveil");
                return 1;
            }
        }
```
?

Reply:
thanks yeah, that's better